packages/signature-controller/src/SignatureController.ts

```diff
@@ -498,6 +544,23 @@ export class SignatureController extends BaseControllerV2<
     );
   }
 
+  #trySetMessageMetadata<
+    M extends AbstractMessage,
+    P extends AbstractMessageParams,
+    PM extends AbstractMessageParamsMetamask,
+  >(
+    messageManager: AbstractMessageManager<M, P, PM>,
```

Comment:
With this change, the `as any` used above is no longer needed.

To preserve these generic parameters without using an `any` cast, we'd need to call this on each type of message manager separately rather than putting them in a list together. That would work too. But I don't see the generics as helping in this case - they don't influence the types used for this method call.

Reply:
I think I prefer this fix, but the counter argument would be that we try to provide full type information whenever possible for the sake of future extensibility where we may add code to return dynamic types or call other methods requiring specific types.

Reply:
Added 👍🏼